### PR TITLE
Initial CI

### PR DESCRIPTION
**Pull request by nrudesu**
Konbanwa! Good Night!
 
I have good news, CI is working (at least in my branch). The CI currently run those script (credits @ammarfaizi2 )
```
sudo apt install -y git gcc g++ make cmake libcurl4-openssl-dev valgrind doxygen;
make -j$(nproc);
make test;
```
Any Improvement or lemme merge to master branch? 

ToDo : 

- [x] Add reporting to CI?

Signed-off-by: Nur Hikari <nrudesu@gmail.com>

### CODE REVIEW

**Comment by ammarfaizi2** [comment]
How do we add report?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        test:
+                runs-on: ubuntu-latest
+                steps:
+                        - uses: actions/checkout@v2
+                        - name : install depend for testing
+                          run : sudo apt install -y git gcc g++ make cmake libcurl4-openssl-dev valgrind doxygen;
+                        - name : build
+                          run : make -j$(nproc);
+                        - name : test
+                          run : make test;
```
Comment (ammarfaizi2) [inline .github/workflows/main.yml:19]: Looks good to me.

**Review by ammarfaizi2 — APPROVED** [review]

**Comment by ammarfaizi2** [comment]
You can merge into master branch. Thank you so much @nrudesu.

**Comment by nrudesu** [comment]
Okay, I merge it.

**Pull request merged**